openedx/core/djangoapps/content/search/api.py

```
@@ -0,0 +1,486 @@
+) -> None:
+    """
+    Creates or updates the document for the given XBlock in the search index
+
+
+    Args:
+        usage_key (UsageKey): The usage key of the XBlock to index
+        recursive (bool): If True, also index all children of the XBlock
+        update_metadata (bool): If True, update the metadata of the XBlock
+        update_tags (bool): If True, update the tags of the XBlock
+    """
+    current_rebuild_index_name = _get_running_rebuild_index_name()
+
+    xblock = modulestore().get_item(usage_key)
+    client = _get_meilisearch_client()
+
+    docs = []
+
+    def add_with_children(block):
+        """ Recursively index the given XBlock/component """
+        doc = searchable_doc_for_course_block(block, include_metadata=update_metadata, include_tags=update_tags)
+        docs.append(doc)
+        if recursive:
+            _recurse_children(block, add_with_children)
+
+    add_with_children(xblock)
+
+    if not docs:
+        return
+
+    tasks = []
+    if current_rebuild_index_name:
+        # If there is a rebuild in progress, the document will also be added to the new index.
+        tasks.append(client.index(current_rebuild_index_name).update_documents(docs))
+    tasks.append(client.index(INDEX_NAME).update_documents(docs))
+
+    _wait_for_meili_tasks(tasks)
+
+
+def delete_index_doc(usage_key: UsageKey) -> None:
+    """
+    Deletes the document for the given XBlock from the search index
+
+    Args:
+        usage_key (UsageKey): The usage key of the XBlock to be removed from the index
+    """
+    current_rebuild_index_name = _get_running_rebuild_index_name()
+
+    client = _get_meilisearch_client()
+
+    tasks = []
+    if current_rebuild_index_name:
+        # If there is a rebuild in progress, the document will also be added to the new index.
+        tasks.append(client.index(current_rebuild_index_name).delete_document(meili_id_from_opaque_key(usage_key)))
```

Will this raise an error if the document doesn't exist in the new index? (We don't want it to - it should just silently ignore if the document doesn't exist.)

---

If the document doesn't exist, it doesn't throw an error. Calling `delete_document` only puts the task in the queue (returning 202).

Getting the task status, we got the following (also, no error):
```
uid=247
index_uid='tutor_studio_content'
status='succeeded'
type='documentDeletion'
details={'providedIds': 1, 'deletedDocuments': 0, 'originalFilter': None}
error=None
canceled_by=None
duration='PT0.002798645S'
enqueued_at=datetime.datetime(2024, 4, 5, 22, 15, 1, 921564)
started_at=datetime.datetime(2024, 4, 5, 22, 15, 1, 923094)
finished_at=datetime.datetime(2024, 4, 5, 22, 15, 1, 925892)
```